src/vs/platform/inputResults/browser/inputResultsService.ts

```diff
@@ -0,0 +1,92 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Event } from 'vs/base/common/event';
+import { IDisposable, combinedDisposable, toDisposable } from 'vs/base/common/lifecycle';
+import { IContextKey, IContextKeyService, RawContextKey } from 'vs/platform/contextkey/common/contextkey';
+import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+
+export const InputResultsViewResultsFocusedContextKey = new RawContextKey<boolean>('inputResultsViewResultsFocused', false);
+export const InputResultsViewInputFocusedContextKey = new RawContextKey<boolean>('inputResultsViewInputFocused', false);
+
+export const IInputResultsService = createDecorator<IInputResultsService>('inputResultsService');
+
+export interface IInputResultsView extends IRegisteredInputResultsView {
+	readonly onDidFocusInput: Event<any>;
+	readonly onDidBlurInput: Event<any>;
+	readonly onDidFocusResults: Event<any>;
+	readonly onDidBlurResults: Event<any>;
+}
+
+export interface IRegisteredInputResultsView {
+	focusInput(): void;
+	focusResults(): void;
+}
+
+export interface IInputResultsService {
+	readonly _serviceBrand: undefined;
+
+	readonly lastFocusedView: IRegisteredInputResultsView | undefined;
+	register(view: IInputResultsView): IDisposable;
+}
+
+export class InputResultsService implements IInputResultsService {
```

Comment:
This is such a strange _thing_ to become a service. First, I don't think we've ever defined such a concept as an _input-result widget_. But let's say we do... still, we don't have a class to encapsulate that widget's definition and behavior. We seem to be creating an abstract concept of a widget without creating the widget itself. 🤔

Reply:
Actually I made it as a service just for defining unified commands and keybindings (i.e., changing the keybinding for one command can affect multiple widgets). But seems like I'm solving an X-Y problem... Do you find it necessary to define unified commands for this navigation?

---

Also after reading other comments you leave, I agree that the term input-results is poorly defined and too constrained. For example in repl,  it's hard to say which one of the filter box and the input box is the "input". Thus I feel like to rewrite this PR and extend it to more general cases -- use Ctrl+Up/Down to navigate among widgets in a vertical-layout container like the search panel did. This should come with the following advantages:

1. No more confusion of "input" vs "results"
2. Can be applied to container with more than two child widgets, instead of simple "input" and "results".

I am wondering what's your attitude to this proposal? Many thanks to your kind instructions.

Reply:
We have what's called a `MultiCommand` - for example https://github.com/microsoft/vscode/blob/c780d0dfef0b84c283e7ca2980491fb3128e4426/src/vs/editor/contrib/clipboard/browser/clipboard.ts#L36-L48

Reply:
> We have what's called a `MultiCommand`

Thanks for pointing out, never know this before.

I am actually rewriting this PR to realize general widget navigation in a container using Ctrl+Up/Down. With MultiCommand I think the service concept can be eliminated. But the time complexity of executing action may increase since there could be many containers attached to the action. Is it acceptable?

Reply:
Yes, that sounds good. Thanks so much for working on this

Reply:
@meganrogge I have another idea to eliminate service, which I'm not sure is appropriate or not. I've put the pseudo code at https://gist.github.com/hsfzxjy/5beae21297024f443c2d69faffc4cfe5 . Could you check it and see whether this pattern is OK or why it's not acceptable?

Reply:
That LGTM